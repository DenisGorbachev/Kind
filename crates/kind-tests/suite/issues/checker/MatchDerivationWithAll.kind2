#derive[match]
type WithCtx (a: Type) { 
    new: Data.U60 -> (WithCtx a)
}