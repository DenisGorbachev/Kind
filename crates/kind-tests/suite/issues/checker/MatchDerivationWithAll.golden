   ERROR  Data constructors cannot return function types.

      /--[suite/issues/checker/MatchDerivationWithAll.kind2:3:10]
      |
    2 |    type WithCtx (a: Type) { 
    3 |        new: Data.U60 -> (WithCtx a)
      |             v----------------------
      |             \Here!
    4 |    }

      Hint: Change all of the function types sequence for explicit arguments like 'cons : x -> T' to 'cons (name: x) : T'

   ERROR  This is not the type that is being declared.

      /--[suite/issues/checker/MatchDerivationWithAll.kind2:2:6]
      |
    2 |    type WithCtx (a: Type) { 
      |         v------
      |         \This is the type that should be used instead
      :     
    3 |        new: Data.U60 -> (WithCtx a)
      |             v----------------------
      |             \This is not the type that is being declared
    4 |    }


